resources/src/asset.rs

```diff
@@ -140,6 +140,31 @@ impl AssetIdentifier {
             _ => Err("Invalid Asset Type.".to_string()),
         }
     }
+
+    /// A type safe way of creating a native asset AssetIdentifier
+    pub fn native() -> AssetIdentifier {
```

Comment:
We should add some simple tests for these. They are super basic but also nice to have.